reconcile/dashdotdb_dvo.py

```diff
@@ -0,0 +1,126 @@
+import logging
+import os
+import requests
+from urllib.parse import urljoin
+
+import reconcile.utils.threaded as threaded
+import reconcile.queries as queries
+from reconcile.utils.oc import OC_Map
+from reconcile.utils.oc import StatusCodeError
+from reconcile.utils.secret_reader import SecretReader
+
+LOG = logging.getLogger(__name__)
+QONTRACT_INTEGRATION = 'dashdotdb-dvo'
+DASHDOTDB_SECRET = os.environ.get('DASHDOTDB_SECRET',
+                                  'app-sre/dashdot/auth-proxy-production')
+
+
+class DashdotdbDVO:
+    def __init__(self, dry_run, thread_pool_size):
+        self.dry_run = dry_run
+        self.thread_pool_size = thread_pool_size
+        self.settings = queries.get_app_interface_settings()
+        secret_reader = SecretReader(settings=self.settings)
+        secret_content = secret_reader.read_all({'path': DASHDOTDB_SECRET})
+        self.dashdotdb_url = secret_content['url']
+        self.dashdotdb_user = secret_content['username']
+        self.dashdotdb_pass = secret_content['password']
+        self.logmarker = "DDDB_DVO:"
+
+    def _post(self, deploymentvalidation):
+        if deploymentvalidation is None:
+            return None
+        cluster = deploymentvalidation['cluster']
+        dvdata = deploymentvalidation['data']
+        LOG.info('%s posting validations for %s', self.logmarker, cluster)
+        response = None
+        if not self.dry_run:
+            endpoint = (f'{self.dashdotdb_url}/api/v1/'
+                        f'deploymentvalidation/{cluster}')
+            response = requests.post(url=endpoint, json=dvdata,
+                                     auth=(self.dashdotdb_user,
+                                           self.dashdotdb_pass))
+            try:
+                response.raise_for_status()
+            except requests.exceptions.HTTPError as details:
+                LOG.error('%s error posting %s - %s',
+                          self.logmarker, cluster, details)
+
+        LOG.info('%s cluster %s synced', self.logmarker, cluster)
+        return response
+
+    def _promget(self, url, query, token=None):
+        uri = 'api/v1/query'
+        url = urljoin('https://'+url, uri)
+        headers = {
+            "accept": "application/json",
+            "Authorization": "Bearer " + token,
+        }
+        params = {'query': query}
+        LOG.info('%s Fetching prom payload from %s?%s',
```

Comment:
I would do this a debug message or this integration logs are going to be very verbose 😅

Reply:
agreed. will change to debug. as one does, littered code for a bit with LOG.info()s during development.. as running with a debug log level was too insanely verbose. development note: this is also how the self.logmarker log prefix came to be, it was easier to filter the debug fire hose that way..

Reply:
changed to debug